packages/jest-snapshot/src/utils.js

```diff
@@ -38,19 +87,32 @@ const getSnapshotPath = (testPath: Path) => path.join(
   path.basename(testPath) + '.' + SNAPSHOT_EXTENSION,
 );
 
-const getSnapshotData = (snapshotPath: Path) => {
+const getSnapshotData = (snapshotPath: Path, update: boolean) => {
   const data = Object.create(null);
+  let snapshotContents;
+  let dirty = false;
 
   if (fileExists(snapshotPath)) {
     try {
-      delete require.cache[require.resolve(snapshotPath)];
-      /* eslint-disable no-useless-call */
-      Object.assign(data, require.call(null, snapshotPath));
-      /* eslint-enable no-useless-call */
+      snapshotContents = fs.readFileSync(snapshotPath, 'utf8');
+      // eslint-disable-next-line no-new-func
+      const populate = new Function('exports', snapshotContents);
+      populate(data);
     } catch (e) {}
   }
 
-  return data;
+  if (!update && snapshotContents) {
+    validateSnapshotVersion(snapshotContents);
+  }
+
+  if (update && snapshotContents) {
+    try {
+      validateSnapshotVersion(snapshotContents);
+    } catch (error) {
```

Comment:
I kinda feel like we could clean this up to return an error and throw in #105 if an error is returned. Here on #109, instead of try-catch, call the validator and set dirty to true if it returns a value. What do you think as a follow-up?